# TC1002S Herramientas computacionales: el arte de la analítica

This is a notebook with all your work for the final evidence of this course


# Niveles de dominio a demostrar con la evidencia

### SING0202A

Interpreta interacciones entre variables relevantes en un problema, como base para la construcción de modelos bivariados basados en datos de un fenómeno investigado que le permita reproducir la respuesta del mismo. Es capaz de construir modelos bivariados que expliquen el comportamiento de un fenómeno.


# Student information

- Name: Rodolfo Arechiga

- ID: A01634610

- My carreer: ITC

# Importing libraries

In [1]:
# Import the packages that we will be using
import pandas as pd                 # For data handling
import seaborn as sns               # For advanced plotting
import matplotlib.pyplot as plt     # For showing plots

# A1 Load data

In [2]:
URL_data = "DatasetsForEvidencia/A01634610.csv"
dataset = pd.read_csv(URL_data)

# A2 Data managment


Print the first 7 rows

In [3]:
dataset.head(7)

Unnamed: 0         x1        x2         x3        x4        x5         x6
0           0  -9.815395 -0.052957  10.266848  2.537235 -7.903850  -8.582304
1           1 -10.986749  1.005359   6.115305  0.515532 -2.270942 -10.482243
2           2 -11.127128  3.555725   9.309544  5.129006 -7.137932  -7.063203
3           3 -10.569800  7.493080   7.182179 -5.795132  0.628242   1.507850
4           4  -6.200259  8.156054  -3.141011 -1.979639 -3.727635  -1.639324
5           5 -10.087523  7.437011  12.384266 -7.054941 -0.951972   4.801125
6           6  -8.738018  5.188146  10.231631 -6.927741 -0.772030   3.609194

Print the first 4 last rows

In [4]:
dataset.tail(4)

Unnamed: 0        x1        x2        x3        x4        x5        x6
606         606 -9.231934  8.347494  8.530035 -7.243500 -4.591418  1.871681
607         607 -8.978481  7.282688  9.078703 -8.201375  1.026061  5.760319
608         608 -6.456217  7.656488 -6.319355 -3.984674 -5.598590 -2.581280
609         609 -4.464305  7.429146 -5.123052 -0.406643 -5.441556 -3.580981

How many rows and columns are in your data?

Use the `shape` method


In [5]:
print(dataset.shape)

(610, 7)


Print the name of all columns 

Use the `columns` method

In [6]:
print(dataset.columns)

Index(['Unnamed: 0', 'x1', 'x2', 'x3', 'x4', 'x5', 'x6'], dtype='object')


What is the data type in each column

Use the `dtypes` method 

In [7]:
print(dataset.dtypes)

Unnamed: 0      int64
x1            float64
x2            float64
x3            float64
x4            float64
x5            float64
x6            float64
dtype: object


What is the meaning of rows and columns?

# Your responses here

 The meaning of rows and columns is in which direction does the contents of the table are, it could be vertical meaning columns and horizontal in te case of the rows inside the file.

Print a statistical summary of your columns


In [8]:
dataset.describe()


Unnamed: 0          x1          x2          x3          x4          x5  \
count  610.000000  610.000000  610.000000  610.000000  610.000000  610.000000   
mean   304.500000   -8.326289    4.758446    4.355848   -1.592809   -3.530693   
std    176.236111    2.445912    2.954964    6.634636    4.004510    2.698017   
min      0.000000  -14.392074   -2.891554   -9.805150   -9.640941   -8.818232   
25%    152.250000  -10.177950    2.143707   -3.581489   -5.028374   -5.595875   
50%    304.500000   -8.776716    5.579284    7.881503   -1.891258   -4.241469   
75%    456.750000   -6.462263    7.026470    9.352455    2.091924   -1.269076   
max    609.000000   -2.049245   12.650762   12.942903    7.314348    4.205866   

               x6  
count  610.000000  
mean    -2.417755  
std      5.041425  
min    -13.189112  
25%     -7.799518  
50%     -1.575657  
75%      2.028723  
max      6.433583

# 1) What is the minumum and maximum values of each variable
    x1 - min = -14.392074, max = -2.049245
    x2 - min = -2.891554, max = 12.650762	
    x3 - min = -9.805150, max = 12.942903
    x4 - min = -9.640941, max = 7.314348
    x5 - min = -8.818232, max = 4.205866	
    x6 - min = -13.189112, max = 6.433583

# 2) What is the mean and standar deviation of each variable
    x1 - mean = -8.326289, standard deviation = 2.445912
    x2 - mean = 4.758446, standard deviation = 2.954964	
    x3 - mean = 4.355848, standard deviation = 6.634636
    x4 - mean = -1.592809, standard deviation = 4.004510
    x5 - mean = -3.530693, standard deviation = 2.698017	
    x6 - mean = -2.417755, standard deviation = 5.041425

# 3) What the 25%, 50% and 75% represent?
    It represents the value at that point between the minimum and the maximum



Rename the columns using the same name with capital letters

In [9]:
rename_columns = dataset.rename(columns = {'x1': 'X1', 'x2': 'X2', 'x3': 'X3', 'x4': 'X4', 'x5': 'X5', 'x6': 'X6'}, inplace = False)

rename_columns

Unnamed: 0         X1        X2         X3        X4        X5         X6
0             0  -9.815395 -0.052957  10.266848  2.537235 -7.903850  -8.582304
1             1 -10.986749  1.005359   6.115305  0.515532 -2.270942 -10.482243
2             2 -11.127128  3.555725   9.309544  5.129006 -7.137932  -7.063203
3             3 -10.569800  7.493080   7.182179 -5.795132  0.628242   1.507850
4             4  -6.200259  8.156054  -3.141011 -1.979639 -3.727635  -1.639324
..          ...        ...       ...        ...       ...       ...        ...
605         605  -6.756404  5.810538  -7.498298  0.114373 -4.765430  -1.384418
606         606  -9.231934  8.347494   8.530035 -7.243500 -4.591418   1.871681
607         607  -8.978481  7.282688   9.078703 -8.201375  1.026061   5.760319
608         608  -6.456217  7.656488  -6.319355 -3.984674 -5.598590  -2.581280
609         609  -4.464305  7.429146  -5.123052 -0.406643 -5.441556  -3.580981

[610 rows x 7 columns]

Rename the columns to their original names

In [10]:
dataset.reindex()

Unnamed: 0         x1        x2         x3        x4        x5         x6
0             0  -9.815395 -0.052957  10.266848  2.537235 -7.903850  -8.582304
1             1 -10.986749  1.005359   6.115305  0.515532 -2.270942 -10.482243
2             2 -11.127128  3.555725   9.309544  5.129006 -7.137932  -7.063203
3             3 -10.569800  7.493080   7.182179 -5.795132  0.628242   1.507850
4             4  -6.200259  8.156054  -3.141011 -1.979639 -3.727635  -1.639324
..          ...        ...       ...        ...       ...       ...        ...
605         605  -6.756404  5.810538  -7.498298  0.114373 -4.765430  -1.384418
606         606  -9.231934  8.347494   8.530035 -7.243500 -4.591418   1.871681
607         607  -8.978481  7.282688   9.078703 -8.201375  1.026061   5.760319
608         608  -6.456217  7.656488  -6.319355 -3.984674 -5.598590  -2.581280
609         609  -4.464305  7.429146  -5.123052 -0.406643 -5.441556  -3.580981

[610 rows x 7 columns]

Use two different alternatives to get one of the columns

In [11]:
print(dataset.loc["x1"])
dataset.drop(['x1', 'x2', 'x4', 'x5', 'x6'], axis='columns', inplace=True)
dataset

KeyError: 'x1'

Get a slice of your data set: second and thrid columns and rows from 62 to 72

For the second and thrid columns, calculate the number of null and not null values and verify that their sum equals the total number of rows

Discard the last column


### Questions 

Based on the previos results, provide a description of yout dataset

Your response:




# A3 Data visualization

Plot in the same figure the histogram of the two variables

Based on this plots, provide a description of your data:

Your response here:

Plot in the same figure the boxplot of the two variables 

Scatter plot of the two variables

### Questions 

Based on the previos plots, provide a description of yout dataset

Your response:




# A4 Kmeans

Do Kmeans clustering assuming a number of clusters accorging to your scatter plot

Add to your dataset a column with the assihned cluster to each data point

Print the number associated to each cluster

Print the centroids

Print the intertia metric

Plot a scatter plot of your data assigned to each cluster. Also plot the centroids

### Questions 

Provides a detailed description of your results

Your response:

# A4 Elbow plot

Compute the Elbow plot

### Questions 

What is the best number of clusters K? (argue your response)

Your response:


Does this number of clusters agree with your inital guess? (argue your response)

Your response:



# BONUS

1) Generate some data using the "make_blobs" function from "sklearn.datasets"

- The number of observations is equal to the three last digits in your ID

- 2 variables

- 4 clusters

- Standar deviation of each cluster of 1.5



2) Plot the scatter plot of your data using the real cluster labels

3) Do K means clustering 

4) Plot the scatter plot of your data using the estimated cluster labels

### Questions 

Provides a detailed description of your results.

Your response:

# Descipcion de tu percepcion del nivel de desarrollo de la subcompetencia 

### SING0202A Interpretación de variables


Escribe tu description del nivel de logro del siguiente criterio de la subcompetencia

**Interpreta interacciones**. Interpreta interacciones entre variables relevantes en un problema, como base para la construcción de modelos bivariados basados en datos de un fenómeno investigado que le permita reproducir la respuesta del mismo.

Tu respuesta:

Escribe tu description del nivel de logro del siguiente criterio de la subcompetencia

**Construcción de modelos**. Es capaz de construir modelos bivariados que expliquen el comportamiento de un fenómeno.

Tu respuesta: